fhir-profile/src/main/java/com/ibm/fhir/profile/ProfileBuilder.java

```diff
@@ -0,0 +1,285 @@
+        for (ElementDefinition elementDefinition : element) {
+            if (elementDefinition.getId().startsWith(prefix)) {
+                elementDefinitions.add(elementDefinition);
+            }
+        }
+        return elementDefinitions;
+    }
+
+    private int indexOf(String id) {
+        for (int i = 0; i < element.size(); i++) {
+            ElementDefinition elementDefinition = element.get(i);
+            if (elementDefinition.getId().equals(id)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public static Binding binding(BindingStrength strength, String valueSet) {
+        return Binding.builder()
+            .strength(strength)
+            .valueSet(Canonical.of(valueSet))
+            .build();
+    }
+
+    public static Binding binding(BindingStrength strength, String valueSet, String maxValueSet) {
+        return Binding.builder()
+            .strength(strength)
+            .valueSet(Canonical.of(valueSet))
+            .extension(Extension.builder()
+                .url("http://hl7.org/fhir/StructureDefinition/elementdefinition-maxValueSet")
+                .value(Canonical.of(maxValueSet))
+                .build())
+            .build();
+    }
+
+    public static Constraint constraint(String key, ConstraintSeverity severity, String human, String expression) {
+        return Constraint.builder()
+                .key(Id.of(key))
+            .severity(severity)
+            .human(string(human))
+            .expression(string(expression))
+            .build();
+    }
+
+    public static Discriminator discriminator(DiscriminatorType type, String path) {
+        return Discriminator.builder()
+            .type(type)
+            .path(string(path))
+            .build();
+    }
+
+    public static Slicing slicing(Discriminator discriminator, SlicingRules rules) {
+        return Slicing.builder()
+            .discriminator(discriminator)
+            .rules(rules)
+            .build();
+    }
+
+    public static List<String> profile(String... profile) {
+        return Arrays.asList(profile);
+    }
+
+    public static List<String> targetProfile(String...targetProfile) {
+        return Arrays.asList(targetProfile);
+    }
+
+    public static Type type(String code) {
+        return type(code, Collections.emptyList(), Collections.emptyList());
+    }
+
+    public static Type type(String code, List<String> profile) {
+        return type(code, profile, Collections.emptyList());
+    }
+
+    public static Type type(String code, List<String> profile, List<String> targetProfile) {
+        return Type.builder()
+            .code(Uri.of(code))
+            .profile(profile.stream()
+                .map(p -> Canonical.of(p))
+                .collect(Collectors.toList()))
+            .targetProfile(targetProfile.stream()
+                .map(p -> Canonical.of(p))
+                .collect(Collectors.toList()))
+            .build();
+    }
+}
```

Comment:
pretty straightforward, however it'd be great to have some degree of javadoc

Reply:
Added javadoc.